app/controllers/application_controller.rb

```diff
@@ -202,7 +190,14 @@ def redirect_back_or_default(default = root_path)
   end
 
   def after_sign_in_path_for(resource)
-    admin_users_path
+    if resource.is_a?(Admin)
+      admin_users_path
+    else
+      back = session[:return_to]
+      session.delete(:return_to)
+
+      back || root_path
```

Comment:
I may be mistaken, but doesn't the existing authlogic code redirect to the user's dashboard on successful login (at least, if it can't find a value in `session[:return_to]`)? Is redirecting to the homepage here a deliberate change?